### PR TITLE
CHAL-23 #done Mock Firebase and test service classes

### PR DESCRIPTION
**Pull request by dvorakjt**
## Checklist
- [x] Include the corresponding Jira issue key and #done in the PR title, like so: "JRA-123 #done Migrate Election Reminders"
- [x] Verify that the code compiles
- [x] Verify that all tests pass
- [x] Verify that unit tests cover 100% of the code
- [ ] <s>Create Storybook stories for visual components</s> N/A
- [ ] <s>Verify that any visual components match the Figma</s> N/A
- [ ] <s>Test with a screen reader (if applicable)</s> N/A
- [x] Document your code with TSDoc comments
- [x] Format your code with Prettier

## Overview
Importing firebase when the jest test environment is set to "jsdom" causes the test suites to fail to run. I could not find a viable solution for this issue except to mock firebase and provide it through the `__mocks__` folder (see https://jestjs.io/docs/manual-mocks). Placing mocks for firebase in this folder causes jest to automatically mock them, circumventing this issue. Mock implementations for all methods/functions/objects exported by firebase and currently used by the project have been provided, but more will need to be added as more features of Firebase are used. These mock implementations store data in memory, so data can be retrieved from them to aid in testing.

I have also provided methods for resetting data stored by auth and firestore between tests. Calling these methods after each test is recommended so as to prevent temporal coupling between tests.

A custom test environment was created that extends jsdom. The FirebaseInviteCodeRepository uses the @paralleldrive/cuid2 package, which depends on TextEncoder. TextEncoder is undefined in jsdom, so this custom environment adds it to the global namespace.

Storybook has been upgraded and `npm audit fix` has been run, eliminated all security warnings. A few warts have been cleaned up.

## Test Plan

See above for the test plan.

## Follow ups

We will need to add to the firebase mocks as we use more features of Firebase. We may also want to consider migrating our tests to Vitest (if the issues with jest mentioned above are not also present in Vitest).

### CODE REVIEW

File: __mocks__/firebase-admin/admin.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    apps.push({
+      getToken: jest.fn(),
+      getCachedToken: jest.fn(),
+      addAuthTokenListener: jest.fn(),
+      removeAuthTokenListener: jest.fn(),
+    });
+  }),
+  credential: {
+    cert: jest.fn().mockImplementation(() => ({
+      getAccessToken: jest.fn,
```
Comment (jyyhuang) [inline __mocks__/firebase-admin/admin.ts]: fn should be a function

**Review by jyyhuang — CHANGES_REQUESTED** [review]
Hi Joseph, this pr looks good. I just noticed a small typing error that should be fixed before merging

File: __mocks__/firebase-admin/admin.ts (hunk 1 above)
Comment (dvorakjt) [reply __mocks__/firebase-admin/admin.ts]: fixed, thank you for catching this!

**Review by jyyhuang — APPROVED** [review]
Approved.

**Pull request merged**